.github/workflows/zmq-lib.yml

```diff
@@ -0,0 +1,176 @@
+
+jobs:
+  get-version:
+    uses: ./.github/workflows/get-version.yml
+
+  package-rpm:
+    needs: [get-version]
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - image: packaging-nfpm-alma8
+            distrib: el8
+          - image: packaging-nfpm-alma9
+            distrib: el9
+
+    runs-on: ubuntu-22.04
+
+    container:
+      image: ${{ vars.DOCKER_INTERNAL_REGISTRY_URL }}/${{ matrix.image }}:${{ needs.get-version.outputs.version }}
+      credentials:
+        username: ${{ secrets.DOCKER_REGISTRY_ID }}
+        password: ${{ secrets.DOCKER_REGISTRY_PASSWD }}
+
+    name: package ${{ matrix.distrib }}
+
+    steps:
+      - name: package rpm
+        run: |
+          dnf install -y wget rpmdevtools rpmlint epel-release
+          dnf config-manager --set-enabled crb || true # alma 9
+          dnf config-manager --set-enabled powertools || true # alma 8
+          dnf install -y asciidoc autoconf automake gcc gcc-c++ glib2-devel libbsd-devel libtool make xmlto
+          cd /github/home
+          wget -O - https://github.com/zeromq/libzmq/archive/refs/tags/v4.3.5.tar.gz|tar zxvf -
+          mkdir -p /github/home/rpmbuild/{BUILD,BUILDROOT,RPMS,SOURCES,SPECS,SRPMS}
+          cp libzmq-4.3.5/packaging/redhat/zeromq.spec /github/home/rpmbuild/SPECS/
+          wget https://github.com/zeromq/libzmq/releases/download/v4.3.5/zeromq-4.3.5.tar.gz -O  /github/home/rpmbuild/SOURCES/zeromq-4.3.5.tar.gz
+          rpmbuild -bb /github/home/rpmbuild/SPECS/zeromq.spec
+        shell: bash
+
+      - name: cache rpm
+        uses: actions/cache/save@13aacd865c20de90d75de3b17ebe84f7a17d57d2 # v4.0.0
+        with:
+          path: /github/home/rpmbuild/RPMS/x86_64/*.rpm
+          key: ${{ github.sha }}-${{ github.run_id }}-rpm-${{ matrix.distrib }}
+
+  package-deb:
+    needs: [get-version]
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - image: packaging-nfpm-bullseye
+            distrib: bullseye
+          - image: packaging-nfpm-bookworm
+            distrib: bookworm
+          - image: packaging-nfpm-jammy
+            distrib: jammy
+
+    runs-on: ubuntu-22.04
+
+    container:
+      image: ${{ vars.DOCKER_INTERNAL_REGISTRY_URL }}/${{ matrix.image }}:${{ needs.get-version.outputs.version }}
+      credentials:
+        username: ${{ secrets.DOCKER_REGISTRY_ID }}
+        password: ${{ secrets.DOCKER_REGISTRY_PASSWD }}
+
+    name: package ${{ matrix.distrib }}
+
+    steps:
+      - name: package deb
+        run: |
+          apt-get update
+          apt-get install -y dpkg-dev wget
+          cd /github/home
+          wget -O - https://github.com/zeromq/libzmq/releases/download/v4.3.5/zeromq-4.3.5.tar.gz|tar zxvf -
+          cd zeromq-4.3.5
+          ./configure
+          make
+          make install
+          cd /github/home
+          wget -O - https://github.com/zeromq/libzmq/archive/refs/tags/v4.3.5.tar.gz|tar zxvf -
```

Comment:
why do you download git tag and then download release ?

Reply:
because we need the release to install zmq locally and the tag to package it